File: caesar.pas
(*Thomas Wojnar*)

Program Caesar;
Uses sysutils;(*Used for AnsiStrUpper(str);*)

VAR enword : String;
    enoffset: Integer;
    deword : String;
	deoffset: Integer;
    solword : String;
	maxshift: Integer;
	
Function Encrypt(input:String; off:Integer) : String;
VAR letter : Integer;
VAR modoff : Integer := off mod 26;
Begin
     FOR letter:=1 TO length(input) DO
		begin
		   if ((ord(input[letter]) >= ord("A")) and (ord(input[letter]) <= ord("Z"))) then
			   begin
				  if ((ord(input[letter]) + modoff) > ord("Z"))
                      then input[letter] := chr((ord(input[letter]) + modoff) - 26)
			      else input[letter] := chr(ord(input[letter]) + modoff);
			   end;
		   end;
	return input;
End;

Function Decrypt(input:String; off:Integer) : String;
VAR letter : Integer;
VAR demodoff : Integer := off mod 26;
Begin
     FOR letter:=1 TO length(input) DO
		begin
		   if ((ord(input[letter]) >= ord("A")) and (ord(input[letter]) <= ord("Z"))) then
			   begin
					    if ((ord(input[letter]) - demodoff) < ord("A"))
		                   then input[letter] := chr((ord(input[letter]) - demodoff) + 26)
					    else input[letter] := chr(ord(input[letter]) - demodoff);
			   end;
		end;
	return input;
End;

Function Solve(input:String; maxshift:Integer) : String;
VAR letter : Integer;
VAR shift : Integer;
Begin
   FOR shift:= 0 TO maxshift DO
   begin
   FOR letter:=1 TO length(input) DO
	  begin
	     if ((ord(input[letter]) >= ord("A")) and (ord(input[letter]) <= ord("Z"))) then
		     begin
			     if ((ord(input[letter]) - 1) < ord("A"))
		            then input[letter] := chr((ord(input[letter]) - 1) + 26)
			     else input[letter] := chr(ord(input[letter]) - 1);
			 end;
		end;
    if shift < 10 then
       Writeln('Caesar 0', shift,': ', input)
    else Writeln('Caesar ', shift,': ', input);
	end;
	return ('done');
End;

Begin
  Write('Enter the String to encrypt: ');
  Readln(enword);
  AnsiStrUpper(enword);
  Write('Enter the offset to encrpyt : ');
  Readln(enoffset); 
  Writeln(ENCRYPT(enword, enoffset));
  Readln;
  
  Write('Enter the String to decrypt: ');
  Readln(deword);
  AnsiStrUpper(deword);
  Write('Enter the offset to decrpyt : ');
  Readln(deoffset);
  Writeln(DECRYPT(deword, deoffset));
  Readln;

  Write('Enter the String to solve: ');
  Readln(solword);
  AnsiStrUpper(solword);
  Write('Enter the max shift amount: ');
  Readln(maxshift);
  Writeln(SOLVE(solword, maxshift));
  Readln;
End.

File: caesar.scala
/**
 * Thomas Wojnar
 */
 
object Main
{
 
  def encrypt (enstr:String, offset:Int) : String = {
    var i:Int = 0
    var modoff = offset % 26
    var edstr = enstr.toCharArray()
    while(i < enstr.length())  {
      if(((enstr(i) toInt) >= ('A' toInt)) && ((enstr(i) toInt) <= ('Z' toInt))){
        if ((enstr.charAt(i) toInt) + modoff > ('Z' toInt)){
          edstr(i) = ((((enstr(i) toInt) + modoff) - 26) toChar)
        }
        else{
          edstr(i) = (((enstr(i) toInt) + modoff) toChar)
        }
      }
      i = i + 1
    }
    val newstr = edstr.mkString("")
    return newstr
  }
  
  def decrypt (enstr:String, offset:Int) : String = {
    var i:Int = 0
    var modoff = offset % 26
    var edstr = enstr.toCharArray()
    while(i < enstr.length())  {
      if(((enstr(i) toInt) >= ('A' toInt)) && ((enstr(i) toInt) <= ('Z' toInt))){
        if ((enstr.charAt(i) toInt) - modoff < ('A' toInt)){
          edstr(i) = ((((enstr(i) toInt) - modoff) + 26) toChar)
        }
        else{
          edstr(i) = (((enstr(i) toInt) - modoff) toChar)
        }
      }
      i = i + 1
    }
    val newstr = edstr.mkString("")
    return newstr
  }
 
 def solve (enstr:String, maxshift:Int) : String = {
    var i = 0
    var j = 0
    var edstr = enstr.toCharArray()
    var newstr = edstr.mkString("")
    while(j <= maxshift) {
    while(i < enstr.length()) {
      if(((enstr(i) toInt) >= ('A' toInt)) && ((enstr(i) toInt) <= ('Z' toInt))){
        if ((enstr.charAt(i) toInt) + j > ('Z' toInt)){
          edstr(i) = ((((enstr(i) toInt) + j ) - 26) toChar)
        }
        else{
          edstr(i) = (((enstr(i) toInt) + j) toChar)
        }
      }
      i = i + 1
        }
        newstr = edstr.mkString("")
        if(j < 10){
            println("Caesar 0" + j.toString + ": " + newstr)
        }
        else{
            println("Caesar " + j.toString + ": " + newstr)
        }
        i = 0
        j = j + 1
        }
    newstr = edstr.mkString("")
    return newstr
  }
  
  def main(args: Array[String]) {
    println("Please input the string you would like to encrypt: ");
    var enstr:String = readLine;
    println("Please input the number you would like to encrypt by: ");
    var en = readInt;
    enstr = enstr.toUpperCase();
    enstr = encrypt(enstr,en)
    println(enstr);
    
        println("Please input the string you would like to decrypt: ");
    var destr:String = readLine;
        println("Please input the number you would like to decrypt by: ");
    var de = readInt;
    destr = destr.toUpperCase();
    destr = decrypt(destr,de);
    println(destr);
    
    println("Please input the string you would like to solve: ");
    var strsol:String = readLine;
        println("Please input the max shift value: ");
    var max = readInt;
    strsol = strsol.toUpperCase();
    strsol = solve(strsol,max);
  }
}